dask/blockwise.py

```diff
@@ -334,6 +378,23 @@ def __dask_distributed_pack__(
         # All blockwise tasks will depend on the futures in `indices`
         global_dependencies = {stringify(f.key) for f in indices_unpacked_futures}
 
+        # Handle `io_deps` serialization.
+        # If `io_deps[<collection_key>]` is just a dict, we rely
+        # entirely on msgpack.  It is up to the `Blockwise` layer to
+        # ensure that all arguments are msgpack serializable. To enable
+        # more control over serialization, a `BlockwiseIODeps` mapping
+        # subclass can be defined with the necessary
+        # `__dask_distributed_{pack,unpack}__` methods.
+        for input_map in self.io_deps.values():
```

Comment:
I don't think this is working as intended, since packed `io_deps` aren't captured. Instead we want something like

```python
        packed_io_deps = {}
        for name, input_map in self.io_deps.items():
            if isinstance(input_map, tuple):
                # Use the `__dask_distributed_pack__` definition for the
                # specified `BlockwiseIODeps` subclass
                io_dep_map = getattr(
                    import_allowed_module(input_map[0]),
                    input_map[1],
                )
                packed_io_deps[name] = io_dep_map.__dask_distributed_pack__(*input_map)
            else:
                packed_io_deps[name] = input_map
```

and to place `packed_io_deps` in the return dictionary

Reply:
oops - good cach, thanks!